lib/cluster/cpg.c

```diff
@@ -398,51 +399,54 @@ pcmk__cpg_message_data(cpg_handle_t handle, uint32_t sender_id, uint32_t pid,
     }
 
     if (handle != 0) {
-        // Do filtering and field massaging
         uint32_t local_nodeid = pcmk__cpg_local_nodeid(handle);
         const char *local_name = pcmk__cluster_local_node_name();
 
-        if ((msg->sender.id != 0) && (msg->sender.id != sender_id)) {
-            crm_err("Nodeid mismatch from %" PRIu32 ".%" PRIu32
-                    ": claimed nodeid=%" PRIu32,
+        // Update or validate message sender ID
+        if (msg->sender.id == 0) {
```

Comment:
In the past, we weren't updating any fields of `msg` until after we had finished validation steps. I'm don't think it would ever matter but I haven't traced closely... suppose we update the sender ID and then fail validation on some other step. Hopefully we haven't contaminated the sender ID in any way by setting it prematurely for a bad message. I think we're fine, or not any worse off than before.

Reply:
Yeah it's fine, this is called by CPG dispatch functions that just ignore the message on NULL returns